test/deploy.js

```diff
@@ -484,6 +485,36 @@ module.exports = (server, token) =>
       });
     });
 
+    tap.test('Should have additional labels', t => {
+      const options = Object.assign(optionsBase, {
+        payload: streamAdditionalLabels,
+      });
+
+      server.inject(options, async response => {
+        // parse result into lines
+        const result = response.result
+          .split('\n')
+          .filter(l => l && l.length)
+          .map(line => JSON.parse(line));
+
+        // find deployments
+        const deployments = result.find(it => it.deployments && it.deployments.length).deployments;
+
+        // check response
+        t.equal(response.statusCode, 200, 'Correct status code');
+
+
+
+        // check docker services
+        const allContainers = await docker.listContainers();
+        const containerInfo = allContainers.find(c => c.Names.includes(deployments[0].Name));
+        t.ok(containerInfo, 'Docker has container');
+        t.equal(containerInfo.Labels['custom.label'], "additional-label", 'Should have label `custom.label=additional-label`');
+        
```

Comment:
Add cleanup, e.g.:
```js
// cleanup
const instance = docker.getContainer(Id);
await instance.stop();
await instance.remove();
```

And it's great 👍

Reply:
Thanks. Was missing that bit!